eessi/testsuite/hooks.py

```diff
@@ -436,10 +436,12 @@ def set_compact_process_binding(test: rfm.RegressionTest):
     # Other launchers may or may not do the correct binding
     test.env_vars['I_MPI_PIN_CELL'] = 'core'  # Don't bind to hyperthreads, only to physcial cores
     test.env_vars['I_MPI_PIN_DOMAIN'] = '%s:compact' % physical_cpus_per_task
-    test.env_vars['OMPI_MCA_rmaps_base_mapping_policy'] = 'node:PE=%s' % physical_cpus_per_task
-    # Default binding for SLURM. Only effective if the task/affinity plugin is enabled
-    # and when number of tasks times cpus per task equals either socket, core or thread count
-    test.env_vars['SLURM_CPU_BIND'] = 'verbose'
+    test.env_vars['OMPI_MCA_rmaps_base_mapping_policy'] = 'slot:PE=%s' % physical_cpus_per_task
+    if test.current_partition.launcher_type().registered_name == 'srun':
+        # Set compact binding for SLURM. Only effective if the task/affinity plugin is enabled
+        # and when number of tasks times cpus per task equals either socket, core or thread count
+        test.env_vars['SLURM_DISTRIBUTION'] = 'block:block'
+        test.env_vars['SLURM_CPU_BIND'] = 'verbose'
```

Comment:
add log for SLURM_DISTRIBUTION here, and move log for SLURM_CPU_BIND in this if block?

Reply:
Done. Also added a log for `I_MPI_PIN_CELL`, since that was missing too. I also nested the `mpirun` variables in a seperate `if`. Additionally, I added a warning message if a launcher is used that we do not support in this function. It's not a blocker, so I won't abort the test, but the user should be aware that performance might be sub-par if the default binding strategy isn't great.